client/core/trade.go

```diff
@@ -574,19 +574,17 @@ func (t *trackedTrade) processCancelMatch(msgMatch *msgjson.Match) error {
 //
 // This method accesses match fields and MUST be called with the trackedTrade
 // mutex lock held for reads.
-func (t *trackedTrade) counterPartyConfirms(match *matchTracker) (have, needed uint32, changed bool) {
+func (t *trackedTrade) counterPartyConfirms(ctx context.Context, match *matchTracker) (have, needed uint32, changed, spent bool) {
 	// Counter-party's swap is the "to" asset.
 	needed = t.wallets.toAsset.SwapConf
 
 	// Check the confirmations on the counter-party's swap.
 	coin := match.counterSwap.Coin()
-	var err error
-	ctx, cancel := context.WithTimeout(context.Background(), confCheckTimeout)
-	defer cancel()
-	have, err = coin.Confirmations(ctx)
+
+	have, spent, err := t.wallets.toWallet.Confirmations(ctx, coin.ID())
```

Comment:
Just a note, no need to change.  I'm really not wild about mixing declaration and assignment with the short variable declaration operator (`:=`), although since `have` and `spent` exist they are assignments, not new variables shadowing the named outputs of `counterPartyConfirms`.  I often go out of my way to declare stuff like `var err error` to avoid the need to analyze the surrounding code to understand what's going on with declaration vs assignment.  Maybe that's just me though.